lib/train/transports/aws.rb

```diff
@@ -0,0 +1,73 @@
+# encoding: utf-8
+require 'train/plugins'
+require 'aws-sdk'
+
+module Train::Transports
+  class Aws < Train.plugin(1)
+    name 'aws'
+    option :region, required: true, default: ENV['AWS_REGION']
+    option :access_key_id, default: ENV['AWS_ACCESS_KEY_ID']
+    option :secret_access_key, default: ENV['AWS_SECRET_ACCESS_KEY']
+    option :session_token, default: ENV['AWS_SESSION_TOKEN']
+
+    # This can provide the access key id and secret access key
+    option :profile, default: ENV['AWS_PROFILE']
+
+    def connection(_ = nil)
+      @connection ||= Connection.new(@options)
+    end
+
+    class Connection < BaseConnection
+      def initialize(options)
+        # Override for any cli options
+        # aws://region/my-profile
+        options[:region] = options[:host] || options[:region]
+        if options[:path]
+          # string the leading / from path
+          options[:profile] = options[:path][1..-1]
+        end
+        super(options)
+
+        @cache_enabled[:api_call] = true
+        @cache[:api_call] = {}
+
+        connect
+      end
+
+      def platform
+        direct_platform('aws')
+      end
+
+      def aws_client(klass)
+        return klass.new unless cache_enabled?(:api_call)
+        @cache[:api_call][klass.to_s.to_sym] ||= klass.new
+      end
+
+      def aws_resource(klass, args)
+        klass.new(args)
+      end
+
+      def connect
+        if @options[:profile]
```

Comment:
What if they have an `.aws/credentials` file with a `[default]` profile set and don't pass that in as the `profile` name here? Will `::Aws::Credentials.new` do the right thing with a bunch of `nil` parameters?

Reply:
Currently if you do not supply a profile it will not default to a default one. It will just look at the env's and if those are nil you will have invalid credentials and get the following error when trying to request something.
`require_credentials': unable to sign request without credentials set (Aws::Errors::MissingCredentialsError)`

We could make it to where we allow aws to do its thing unless env's or a profile is set. This would allow you to use the default creds. To do this we would just want to update line 60 to ~
```
        opts = { region: @options[:region] }
        opts[:credentials] = creds unless creds.access_key_id.nil?
```
This will keep the default credentials unless we are actually updating them.

Reply:
I could’ve sworn that if you set up a client object and don’t set any credentials object in the config, it just does The Right Thing with regards to reading env vars, or reading the credentials file and using the “default” group by default.

Reply:
Thanks - I think that's the right call so that people have a consistent UX with all things that use the AWS SDK, including Train.